src/main/java/com/streisky/virtualshowcasebackend/controller/product/ProductController.java

```diff
@@ -31,61 +30,37 @@ public class ProductController {
 
     @PostMapping
     @Transactional
-    public ResponseEntity<ProductDTO> register(@RequestBody @Valid ProductDTO productDTO) {
-        try {
-            ProductDTO savedProduct = productService.save(productDTO);
-            URI location = ServletUriComponentsBuilder
-                    .fromCurrentRequest()
-                    .path("/{id}")
-                    .buildAndExpand(savedProduct.id())
-                    .toUri();
+    public ResponseEntity<ProductDTO> save(@RequestBody @Valid ProductDTO productDTO) {
+        ProductDTO savedProduct = productService.save(productDTO);
+        URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest()
+                .path("/{id}")
+                .buildAndExpand(savedProduct.id())
+                .toUri();
 
-            return ResponseEntity.created(location).body(savedProduct);
-        } catch (Exception e) {
-            return ResponseEntity.badRequest().build();
-        }
+        return ResponseEntity.created(location).body(savedProduct);
     }
 
     @PutMapping
     @Transactional
     public ResponseEntity<ProductDTO> update(@RequestBody @Valid ProductDTO productDTO) {
-        try {
-            return ResponseEntity.ok(productService.save(productDTO));
-        } catch (Exception e) {
-            return ResponseEntity.badRequest().build();
-        }
+        return ResponseEntity.ok(productService.update(productDTO));
     }
 
     @DeleteMapping("/{id}")
     @Transactional
     public ResponseEntity<Void> delete(@PathVariable Long id) {
-        try {
-            productService.delete(id);
-            return ResponseEntity.ok().build();
-        } catch (VirtualShowcaseNotFoundException e) {
-            return ResponseEntity.notFound().build();
-        } catch (Exception e) {
-            return ResponseEntity.badRequest().build();
-        }
+        productService.delete(id);
+        return ResponseEntity.noContent().build();
     }
 
     @GetMapping("/{id}")
     public ResponseEntity<ProductDTO> find(@PathVariable Long id) {
-        try {
-            return ResponseEntity.ok(productService.find(id));
-        } catch (VirtualShowcaseNotFoundException e) {
-            return ResponseEntity.notFound().build();
-        } catch (Exception e) {
-            return ResponseEntity.badRequest().build();
-        }
+        return ResponseEntity.ok(productService.find(id));
     }
 
     @GetMapping
     public ResponseEntity<Page<ProductDTO>> findAll(@PageableDefault(size = 10, sort = "description") Pageable pageable) {
```

Comment:
A suggestion for upcoming development is to let the client decide page size and the sorting parameter.

Reply:
The client can decide the size. This is a default value that is overwritten if values are entered in the parameters.

Example:
`/products?page=0&size=20&sort=id`